crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -152,44 +161,52 @@ impl FunctionBuilder {
     }
 
     /// Switch to inserting instructions in the given block.
+    ///
     /// Expects the given block to be within the same function. If you want to insert
-    /// instructions into a new function, call new_function instead.
+    /// instructions into a new function, call `new_function` instead.
     pub(crate) fn switch_to_block(&mut self, block: BasicBlockId) {
         self.current_block = block;
     }
 
-    /// Returns the block currently being inserted into
+    /// Returns the block currently being modified
     pub(crate) fn current_block(&mut self) -> BasicBlockId {
         self.current_block
     }
 
-    /// Insert an allocate instruction at the end of the current block, allocating the
-    /// given amount of field elements. Returns the result of the allocate instruction,
+    /// Insert an allocate instruction at the end of the current block.
+    /// Returns the result of the allocate instruction,
     /// which is always a Reference to the allocated data.
     pub(crate) fn insert_allocate(&mut self) -> ValueId {
+        // TODO: Rust has .first on vectors which is confusing because
+        // TODO that returns Option<T> whereas this returns T and expects there
+        // TODO to be only one T.Perhaps change this to be `first_and_only`
+        // TODO or something to not conflate this with .first, but also
+        // TODO to convey the fact that its the only one; `exactly_one` ?
         self.insert_instruction(Instruction::Allocate, None).first()
     }
 
-    /// Insert a Load instruction at the end of the current block, loading from the given offset
-    /// of the given address which should point to a previous Allocate instruction. Note that
-    /// this is limited to loading a single value. Loading multiple values (such as a tuple)
-    /// will require multiple loads.
-    /// 'offset' is in units of FieldElements here. So loading the fourth FieldElement stored in
-    /// an array will have an offset of 3.
-    /// Returns the element that was loaded.
+    /// Insert a Load instruction at the end of the current block.
+    /// TODO: check previous description was not outdated.
+    /// TODO: where do we check that `address` is a Reference?
```

Comment:
The previous doc comment was certainly outdated.

We don't check whether the given address is a `Type::Reference`. We don't perform type checking in the SSA in general.